lib/portage/dbapi/bintree.py

```diff
@@ -1087,18 +1095,20 @@ def _populate_local(self, reindex=True):
                         msg = []
                         if missing_keys:
                             missing_keys.sort()
+                            if invalid_errors:
+                                msg.append(
+                                    _("Missing metadata key(s): %s.")
+                                    % ", ".join(missing_keys)
+                                )
+                        if invalid_errors:
```

Comment:
We could move the above msg variable definition below the invalid_errors condition like this:
```python
if invalid_errors:
    msg = []
    if missing_keys:
        missing_keys.sort()
        msg.append(
```